docs/operating-eck/troubleshooting/common-problems.asciidoc

```diff
@@ -84,3 +84,75 @@ type: Opaque
 ----
 
 Failure to do so can cause data loss.
+
+[id="{p}-{page_id}-pod-updates"]
+== Pods are not replaced after a configuration update
+
+The update of an existing Elasticsearch cluster configuration can fail because the operator is unable to apply the changes required while replacing the pods of a given Elasticsearch cluster.
+
+This can be identified by the `AGE` of the pod being older than the configuration update, or the `STATUS` of the pods:
+
+[source,sh]
+----
+kubectl get pods
+
+NAME                      READY   STATUS    RESTARTS   AGE
+quickstart-es-default-0   1/1     Running   0          15m
+quickstart-es-default-1   1/1     Running   0          15m
+quickstart-es-default-2   1/1     Running   0          15m
+----
+
+The causes include:
+
+- The Elasticsearch cluster is not healthy.
+
+The health of the cluster may look like
+
+[source,sh]
+----
+kubectl get elasticsearch
+
+NAME         HEALTH   NODES   VERSION   PHASE   AGE
+TODO: Finish example
+----
+
+TODO:
+Action: Check and fix your shard allocations.
+(more API details can be added here)
+For more information, see link:https://www.elastic.co/guide/en/elasticsearch/reference/7.10/cluster-health.html[Cluster health API]
+and link:https://www.elastic.co/guide/en/cloud-on-k8s/1.3/k8s-deploy-elasticsearch.html#k8s_monitor_cluster_health_and_creation_progress[Monitor cluster health and creation progress].
+
+- Not enough resources in the Kubernetes cluster
+
+The scheduling might fail with the following message:
+
+[source,sh]
+----
+kubectl get events --sort-by='{.lastTimestamp}' | tail
+
+LAST SEEN   TYPE      REASON             OBJECT                        MESSAGE
+10s         Warning   FailedScheduling   pod/quickstart-es-default-2   0/3 nodes are available: 3 Insufficient memory.
+----
+
+Or a pod is stuck in `STATUS` of `Pending` with an operator log of unable to `restart some nodes`
+
+[source,sh]
+----
+kubectl get pods
+
+NAME                      READY   STATUS    RESTARTS   AGE
+quickstart-es-default-0   1/1     Running   0          146m
+quickstart-es-default-1   1/1     Running   0          146m
+quickstart-es-default-2   0/1     Pending   0          134m
+
+
+kubectl -n elastic-system logs statefulset.apps/elastic-operator | tail
+
+{"log.level":"info","@timestamp":"2020-11-19T17:34:48.769Z","log.logger":"driver","message":"Cannot restart some nodes for upgrade at this time","service.version":"1.3.0+6db1914b","service.type":"eck","ecs.version":"1.4.0","namespace":"default","es_name":"quickstart","failed_predicates":{"do_not_restart_healthy_node_if_MaxUnavailable_reached":["quickstart-es-default-1","quickstart-es-default-0"]}}
+----
```

Comment:
```suggestion
The operator may log it is unable to `restart some nodes`:

[source,sh]
----
kubectl -n elastic-system logs statefulset.apps/elastic-operator | tail

{"log.level":"info","@timestamp":"2020-11-19T17:34:48.769Z","log.logger":"driver","message":"Cannot restart some nodes for upgrade at this time","service.version":"1.3.0+6db1914b","service.type":"eck","ecs.version":"1.4.0","namespace":"default","es_name":"quickstart","failed_predicates":{"do_not_restart_healthy_node_if_MaxUnavailable_reached":["quickstart-es-default-1","quickstart-es-default-0"]}}
----

A pod is stuck in `STATUS` of `Pending`:

[source,sh]
----
kubectl get pods

NAME                      READY   STATUS    RESTARTS   AGE
quickstart-es-default-0   1/1     Running   0          146m
quickstart-es-default-1   1/1     Running   0          146m
quickstart-es-default-2   0/1     Pending   0          134m
----
```

Reply:
I did not figure out is _where_ the `kubectl` command would tell me "why?" the pod is stuck in pending.  As I purposefully set-up the failure, I knew the lack of compute resources was the cause but was not able to find message to tell me this is the case.  I'm sure the info is somewhere, I just don't know where to look or had the time to look into it.

Reply:
@sebgl asking for your help here.

Reply:
> I did not figure out is where the kubectl command would tell me "why?" the pod is stuck in pending

What you're looking for are `events` attached to that Pod. They live in a different structure: the `events` (`kubectl get events`), not the Pod.

Fortunately, there's an easy shortcut to grab all events attached to a particular Pod:

```
> kubectl describe pod <pod-name>

(... a lot of stuff here including the pod spec.... but at the very end of the output you'll see the events:)

Events:
  Type     Reason             Age                From                Message
  ----     ------             ----               ----                -------
  Normal   NotTriggerScaleUp  91s                cluster-autoscaler  pod didn't trigger scale-up (it wouldn't fit if a new node is added):
  Warning  FailedScheduling   14s (x4 over 93s)  default-scheduler   0/3 nodes are available: 3 Insufficient memory.
```